lib/Transforms/Scalar/DxilLoopUnroll.cpp

```diff
@@ -450,7 +457,12 @@ static void FindProblemBlocks(BasicBlock *Header, const SmallVectorImpl<BasicBlo
 
     for (User *U : I->users()) {
       if (Instruction *UserI = dyn_cast<Instruction>(U)) {
-        WorkList.push_back(UserI);
+        if (!InstructionsSeen.count(UserI) &&
+          BlocksInLoopSet.count(UserI->getParent()))
+        {
+          InstructionsSeen.insert(UserI);
```

Comment:
Oops, I don't think you want to do this.  Otherwise, as soon as you pop it off the worklist, you'll think you've already processed it and skip it.

Reply:
Oh, I see you removed the code that checked whether it was still in the worklist after popping it off. This is probably ok, but the original PHINodes weren't added to InstructionsSeen, right? Should probably do that now.